ccmlib/scylla_node.py

```diff
@@ -1785,6 +1785,78 @@ def wait_for_compactions(self,
                                      column_family=column_family,
                                      idle_timeout=idle_timeout)
 
+    def run_scylla_types(self, action: str, values: list[str], types: list[str], additional_args: Optional[list[str]]=None, text=True):
+        """Invoke scylla-types command, with the specified action and additional_args.
+
+        For more information about scylla-types, see https://opensource.docs.scylladb.com/stable/operating-scylla/admin-tools/scylla-types.html
+
+        Params:
+        * action - The scylla-types operation to run.
+        * additional_args - Additional command-line arguments to pass to scylla-types action, this should be a list of strings.
+        * text - If True, output of the command is treated as text, if not as bytes
+
+        Returns: stdout, stderr
+
+        Raises: subprocess.CalledProcessError if scylla-types returns a non-zero exit code.
+        """
+        if additional_args is None:
+            additional_args = []
+
+        for value_type in types:
+            additional_args.extend(["-t", value_type])
+
+        scylla_path = common.join_bin(self.get_path(), BIN_DIR, 'scylla')
+        scylla_types_cmd_with_args = [scylla_path, "types", action] + additional_args + ["--"] + values
+        res = subprocess.run(scylla_types_cmd_with_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=text, check=False, env=self._get_environ())
+        if res.returncode:
+            raise ToolError(command=' '.join(scylla_types_cmd_with_args), exit_status=res.returncode, stdout=res.stdout, stderr=res.stderr)
+        return (res.stdout, res.stderr)
+
+    def serialize_values(self, values: list[str], types: list[str], prefixable_compounds=False) -> str:
+        """Serialize the given values using scylla-types serialize action
+
+        Params:
+        * values : the values to be serialized
+        * types : cassandra type class name of the values' types (example : ShortType, Int32Type, TimeUUIDType, etc.)
+        * prefixable_compounds: if true, values are prefixable compounds (e.g. clustering key)
+                                else, the values are full compounds (e.g. partition keys)
+
+        Returns: serialized value
+
+        Raises: subprocess.CalledProcessError if scylla-types serialize returns a non-zero exit code.
+
+        """
+        if len(values) > len(types):
```

Comment:
What about using a dict? That way it cannot get mismatched

Reply:
@pehala  We cannot use a dict because we need to allow duplicate keys.

I tried using List[Tuple[str, str]], but if prefixable_compounds is set to True, the length of values can be less than the length of types and the existing code seemed more user friendly for such cases. So, I have left it as it is.